internal/services/storage/storage_account_resource.go

```diff
@@ -458,6 +458,11 @@ func resourceStorageAccount() *pluginsdk.Resource {
 										Default:      7,
 										ValidateFunc: validation.IntBetween(1, 365),
 									},
+									"allow_permanent_delete": {
```

Comment:
Can we rename this to `permanent_delete_enabled` (here and also the document) to be consistent to the naming convention among the provider?